app/components/Article/index.tsx

```diff
@@ -148,6 +151,51 @@ const ArticleFooter = (question: Question) => {
       month: 'short',
     })
 
+  const actionIds = {helpful: false, unhelpful: false}
+
+  const loadActionTaken = () => {
+    forEach(actionIds, (value, key) => {
+      try {
+        actionIds[key] = localStorage.getItem(`${question.pageid}-${key}`) === 'true'
+      } catch (e) {
+        // This will happen when local storage is disabled
+        actionIds[key] = false
+      }
+    })
+  }
+
+  const actionParams = (actionType) => {
+    return new URLSearchParams({
+      pageid: question.pageid,
+      actionTaken: actionIds[actionType].toString(),
+      action: actionType,
+    })
+  }
+  const switchAction = (actionType) => {
+    if (actionIds[actionType]) {
+      actionIds[actionType === 'helpful' ? 'unhelpful' : 'helpful'] = false
+    }
+  }
+  const handleAction = async (actionTaken) => {
+    actionIds[actionTaken] = !actionIds[actionTaken]
+    switchAction(actionTaken)
+    const searchParams = actionParams(actionTaken)
+    const response = await fetch('/questions/actions', {method: 'POST', body: searchParams})
+
+    if (response.ok !== true) {
+      actionIds[actionTaken] = !actionIds[actionTaken]
+      switchAction(actionTaken)
+      return
+    }
+    try {
+      forEach(actionIds, (value, key) => {
```

Comment:
why are you always setting all actions, rather than just the current action?

Reply:
Because they change values.. an article cannot be helpful and unhelpful simultaneously.

Reply:
why not? It absolutely can :D